tests/lib/rules/prefer-regex-literals.js

```diff
@@ -134,123 +129,970 @@ ruleTester.run("prefer-regex-literals", rule, {
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+
+        {
+            code: "new RegExp('+');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "new RegExp('*');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('+');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('*');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "new RegExp('+', 'g');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "new RegExp('*', 'g');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('+', 'g');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('*', 'g');",
+            output: null,
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('abc', 'u');",
+            output: null,
+            parserOptions: { ecmaVersion: 3 },
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('abc', 'd');",
+            output: null,
+            parserOptions: { ecmaVersion: 2021 },
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('abc', 'd');",
+            output: "/abc/d;",
+            parserOptions: { ecmaVersion: 2022 },
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\\\\\\\', '');",
+            output: "/\\\\/;",
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\n', '');",
+            output: "/\\n/;",
+            parserOptions: { ecmaVersion: 2022 },
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\n\\n', '');",
+            output: "/\\n\\n/;",
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\t', '');",
+            output: "/\\t/;",
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\t\\t', '');",
+            output: "/\\t\\t/;",
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\r\\n', '');",
+            output: "/\\r\\n/;",
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
+        {
+            code: "RegExp('\\u1234', 'g')",
+            output: "/ሴ/g",
+            errors: [{ messageId: "unexpectedRegExp" }]
+        },
```

Comment:
This is also problematic as the user's intent is to have `\u1234` in the code, not `ሴ`.

There are also characters that almost never appear in a source code, e.g. `RegExp('\0')` shouldn't be autofixed to a pattern with the NULL character.

We could maybe limit the autofix to work with patterns that contain a predefined set of characters only.

Reply:
> We could maybe limit the autofix to work with patterns that contain a predefined set of characters only.

Yeah, that is the only apparent solution because, we now use `getStaticValue`.

Reply:
Why do we need getStaticValue?

Reply:
To understand code like `'a' + 'b'` as static.

Reply:
> To understand code like `'a' + 'b'` as static.

That isn't in the scope of https://github.com/eslint/eslint/issues/15029 proposal. We're only evaluating whether it is possible to autofix existing errors, not whether this rule should be reporting more errors.

Reply:
Actually, now that I remember, the reason that I made this was because strings such as 
```js
String.raw`\d`
```
were not able to be evaluated to `'\\d'`. Should I create a new proposal and link it to this pull request?

Reply:
> ```js
> String.raw`\d`
> ```

If this is `node` (TaggedTemplateExpression), we can get the value like this:

```js
node.quasi.quasis[0].value.raw
```